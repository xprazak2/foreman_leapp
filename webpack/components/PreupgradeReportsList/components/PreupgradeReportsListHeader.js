import React from 'react';
import { ListView } from 'patternfly-react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import InfoItem from './InfoItem';
import SortableHeaderItem from './SortableHeaderItem';

const PreupgradeReportsListHeader = ({ sort, changeSort, toggleSelectAll }) => {
  const additionalInfoColumns = [
    { label: 'Host', value: 'hostname' },
    { label: 'Severity', value: 'severity' },
    { label: 'Has Remediation?', value: 'fix' },
  ];

  const additionalInfo = additionalInfoColumns.map((col, idx) => (
    <InfoItem itemId={`${idx}`} key={idx}>
      <strong>
        <SortableHeaderItem
          title={col.label}
          sort={sort}
          value={col.value}
          changeSort={changeSort}
        />
      </strong>
    </InfoItem>
  ));

  return (
    <ListView.Item
      additionalInfo={additionalInfo}
      className={classNames(
        'listViewItem--listItemVariants',
        'list-view-header'
      )}
      checkboxInput={<input type="checkbox" onClick={toggleSelectAll} />}
      description={
        <SortableHeaderItem
          title="Title"
          sort={sort}
          value="title"
          changeSort={changeSort}
        />
      }
      hideCloseIcon
      stacked
    />
  );
};

PreupgradeReportsListHeader.propTypes = {
  sort: PropTypes.object.isRequired,
  changeSort: PropTypes.func.isRequired,
  toggleSelectAll: PropTypes.func.isRequired,
};

export default PreupgradeReportsListHeader;
