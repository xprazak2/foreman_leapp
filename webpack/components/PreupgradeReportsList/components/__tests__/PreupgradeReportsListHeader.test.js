import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import PreupgradeReportsListHeader from '../PreupgradeReportsListHeader';

const fixtures = {
  'should render': {
    sort: { attribute: '', order: 'asc' },
    changeSort: () => {},
    toggleSelectAll: () => {},
  },
};

describe('PreupgradeReportsListHeader', () =>
  testComponentSnapshotsWithFixtures(PreupgradeReportsListHeader, fixtures));
