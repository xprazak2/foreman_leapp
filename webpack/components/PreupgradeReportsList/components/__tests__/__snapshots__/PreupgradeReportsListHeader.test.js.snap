// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreupgradeReportsListHeader should render 1`] = `
<ListViewItem
  actions={null}
  additionalInfo={
    Array [
      <InfoItem
        itemId="0"
        tooltipText=""
      >
        <strong>
          <SortableHeaderItem
            changeSort={[Function]}
            sort={
              Object {
                "attribute": "",
                "order": "asc",
              }
            }
            title="Host"
            value="hostname"
          />
        </strong>
      </InfoItem>,
      <InfoItem
        itemId="1"
        tooltipText=""
      >
        <strong>
          <SortableHeaderItem
            changeSort={[Function]}
            sort={
              Object {
                "attribute": "",
                "order": "asc",
              }
            }
            title="Severity"
            value="severity"
          />
        </strong>
      </InfoItem>,
      <InfoItem
        itemId="2"
        tooltipText=""
      >
        <strong>
          <SortableHeaderItem
            changeSort={[Function]}
            sort={
              Object {
                "attribute": "",
                "order": "asc",
              }
            }
            title="Has Remediation?"
            value="fix"
          />
        </strong>
      </InfoItem>,
    ]
  }
  checkboxInput={
    <input
      onClick={[Function]}
      type="checkbox"
    />
  }
  className="listViewItem--listItemVariants list-view-header"
  compoundExpand={false}
  compoundExpanded={false}
  description={
    <SortableHeaderItem
      changeSort={[Function]}
      sort={
        Object {
          "attribute": "",
          "order": "asc",
        }
      }
      title="Title"
      value="title"
    />
  }
  heading={null}
  hideCloseIcon={true}
  initExpanded={false}
  leftContent={null}
  onCloseCompoundExpand={[Function]}
  onExpand={[Function]}
  onExpandClose={[Function]}
  stacked={true}
/>
`;
